### DISCUSSION TITLE
Fresnel Lens ?

### DISCUSSION THREAD

**Discussion by mikamyara** [mjhoptics/ray-optics#98] [Q&A] [answered]
Hello everyone,

   I am very new (2-hours !)  user of the ray-optics library. Seems to be so nice ! 
   I asked a question to myself : does it make sens to simulate a Fresnel Lens with this kind of library ? And if it makes sense, how could I do it ? can I do it with "lots" of add_surface calls ? 

  Regards,
     Mike

**Comment by mjhoptics** [comment, marked as answer]
Hello Mike,

A Fresnel lens such as you describe cannot be modeled currently in **ray-optics**. All of the fresnel zones have to be considered simultaneously; using add_surface would add each zone in sequence which is not what you want.

There is a "simplified" model of a fresnel lens that has been included in optical design programs for many years. The "fresnel" part of the surface specification perturbs the surface normal continuously across the surface. This perturbation is on top of a surface profile description. It can be useful in some scenarios. This is not currently implemented in **ray-optics**, but would be considerably less work than the full surface model.

HTH
Mike Hayford

**Reply by mikamyara** [reply]
Ok thanks a lot for your answer !
